t/test_getArgsForCall

```diff
@@ -0,0 +1,102 @@
+#!/bin/sh
+
+readonly TEST_DIR="$(dirname "$0")"
+
+# shellcheck source=/dev/null
+. "$TEST_DIR/../shpy"
+
+testCalledWithNoArgs_ShouldReturnUsage() {
+    # shellcheck disable=SC2039
+    local output
+
+    output=$(getArgsForCall 2>&1 >/dev/null)
+
+    assertFalse "\`getArgsForCall\` is false" $?
+    printf '%s\n' "$output" | grep -q '^Usage:'
+    assertTrue 'output begins with "Usage:"' $?
+}
+
+testCalledWithOneArg_ShouldReturnUsage() {
+    # shellcheck disable=SC2039
+    local output
+
+    output=$(getArgsForCall some_spy 2>&1 >/dev/null)
+
+    assertFalse "\`getArgsForCall some_spy\` is false" $?
+    printf '%s\n' "$output" | grep -q '^Usage:'
+    assertTrue 'output begins with "Usage:"' $?
+}
+
+testCalledWithInvalidSpy_ShouldReturnError() {
+    # shellcheck disable=SC2039
+    local output
+
+    output=$(getArgsForCall invalid_spy 1 2>&1 >/dev/null)
+
+    assertFalse "\`getArgsForCall invalid_spy 1\` is false" $?
+    printf '%s\n' "$output" | grep -q '^Error:'
+    assertTrue 'output begins with "Error:"' $?
```

Comment:
I see what you're saying, but I did it this way for two reasons:

1. Maintain consistency with existing tests that do something similar (test_examineNextSpyCall, test_shunit2Interface, and test_wasSpyCalledWith)
2. Not tightly coupling the test to the unit under test.  The main goal of this test is to check that an error is "thrown", not that it has exactly the right error message.  If we test for the whole message, then when that message changes in the future, we have to ensure the test changes with it.

That being said, this is a fairly simple test, so I'm happy to have it look for the whole error message.  Let me know.

Reply:
That's a good point. I think it's good to verify the correct error is thrown, since seeing the wrong message would be confusing, even if it means the tests are more coupled.

Reply:
OK I've changed the test to look for the whole error message